<?xml version="1.0"?>
<robot name="mobile_robot" xmlns:xacro="http://ros.org/wiki/xacro">

    <material name="blue">
	    <color rgba="0 0 0.8 1"/>
    </material>

    <material name="white">
	    <color rgba="1 1 1 1"/>
    </material>

    <material name="black">
	    <color rgba="0 0 0 1"/>
    </material>

    <material name="green">
        <color rgba="0.0 0.8 0.0 1.0"/>
    </material>

    <material name="orange">
        <color rgba="1 0.3 0.1 1"/>
    </material>

    <material name="red">
        <color rgba="1 0 0 1"/>
    </material>

    <link name="base_link">
        <!--<visual>
          <geometry>
             <box size="0.24 0.24 0.001"/>
          </geometry>
          <material name="white"/>
          <origin rpy="0 0 0" xyz="0 0 0.0005"/>
          <material>
            <color rgba="0 0 0.8 1" />
          </material>
        </visual> -->
        <collision>
          <origin rpy="0 0 0"/>
          <geometry>
            <box size="0.49 0.41 0.83"/>
          </geometry>
        </collision>
    </link>

    <joint name="base_link_joint" type="fixed">
        <origin xyz="-0.1 0 0"/>
        <parent link="base_link" />
        <child link="base_plate" />
    </joint>

    <link name="base_plate">
    <visual>
      <origin xyz="0.53 0.213 -0.1" rpy="0 0 ${pi}"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/main_body.stl" scale="0.001 0.001 0.001" /> 
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <origin xyz="0.53 0.213 -0.1" rpy="0 0 ${pi}"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/main_body.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>   
    
    <!-- RIGHT WHEEL LINK -->
    <joint name="right_wheel_joint" type="continuous">
        <origin xyz="0 -0.23 0" rpy="${pi/2} 0 0" />
        <parent link="base_plate" />
        <child link="right_wheel" />
        <axis xyz="0 0 -1"/>
    </joint>

    <link name="right_wheel">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/wheel.stl" scale="0.001 0.001 0.001"/> 
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/wheel.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>   
    
    <!-- LEFT WHEEL LINK -->
    <joint name="left_wheel_joint" type="continuous">
        <origin xyz="0 0.23 0" rpy="0 ${pi} ${pi/2}"/>
        <parent link="base_plate" />
        <child link="left_wheel" />
        <axis xyz="-1 0 0" />
    </joint>

    <link name="left_wheel">
    <visual>
      <origin xyz="0 0 0" rpy="${pi} -${pi/2} 0"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/wheel.stl" scale="0.001 0.001 0.001"/> 
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="${pi} -${pi/2} 0"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/wheel.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>   
    
    <!-- KINECT LINK -->
    <joint name="laser_joint" type="fixed">
        <origin xyz="-0 0 0.83" rpy="0 0 0"/>
        <parent link="base_plate" />
        <child link="camera_link" />
    </joint>

    <link name="camera_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 ${pi}"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/kinect.stl" scale="0.001 0.001 0.001"/> 
      </geometry>
      <material name="orange"/>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 ${pi}"/>
      <geometry>
        <mesh filename="package://autonomus_mobile_robot/stl/kinect.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>  
    
</robot>
